pkg/lib/database/filter.go

```diff
@@ -845,3 +868,91 @@ func (i *FilterNestedNotIn) AnystoreFilter() query.Filter {
 func (i *FilterNestedNotIn) IterateNestedFilters(fn func(nestedFilter Filter) error) error {
 	return fn(i)
 }
+
+type Filter2ValuesComp struct {
+	Key1, Key2 string
+	Cond       model.BlockContentDataviewFilterCondition
+}
+
+func (i Filter2ValuesComp) FilterObject(g *types.Struct) bool {
+	val1 := pbtypes.Get(g, i.Key1)
+	val2 := pbtypes.Get(g, i.Key2)
+	eq := FilterEq{Value: val2, Cond: i.Cond}
+	return eq.filterObject(val1)
+}
+
+func (i Filter2ValuesComp) AnystoreFilter() query.Filter {
+	var op query.CompOp
+	switch i.Cond {
+	case model.BlockContentDataviewFilter_Equal:
+		op = query.CompOpEq
+	case model.BlockContentDataviewFilter_Greater:
+		op = query.CompOpGt
+	case model.BlockContentDataviewFilter_GreaterOrEqual:
+		op = query.CompOpGte
+	case model.BlockContentDataviewFilter_Less:
+		op = query.CompOpLt
+	case model.BlockContentDataviewFilter_LessOrEqual:
+		op = query.CompOpLte
+	case model.BlockContentDataviewFilter_NotEqual:
+		op = query.CompOpNe
+	}
+	return &Anystore2ValuesComp{
+		Path1:  []string{i.Key1},
+		Path2:  []string{i.Key2},
+		CompOp: op,
+	}
+}
+
+type Anystore2ValuesComp struct {
+	Path1, Path2 []string
+	CompOp       query.CompOp
+	buf1, buf2   []byte
+}
+
+func (e *Anystore2ValuesComp) Ok(v *fastjson.Value) bool {
+	value1 := v.Get(e.Path1...)
+	value2 := v.Get(e.Path2...)
+	comp := bytes.Compare(encoding.AppendJSONValue(e.buf1[:0], value1), encoding.AppendJSONValue(e.buf2[:0], value2))
```

Comment:
e.buf is always nil, you should set it to a new value

```
e.buf = encoding.AppendJSONValue(e.buf1[:0], value1)
```

Reply:
Thank you! Added init of each buffer before comparison